src/App.tsx

```diff
@@ -1,26 +1,187 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, deleteTodo, getTodos, postTodo } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { HeaderInput } from './components/HeaderInput';
+import { Footer } from './components/Footer';
+import classNames from 'classnames';
+import { Errors, SelectedTasks, Todo } from './types/Types';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [selectedTasks, setSelectedTasks] = useState<SelectedTasks>(
+    SelectedTasks.All,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>();
+  const [title, setTitle] = useState('');
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [isAddingTodo, setIsAddingTodo] = useState(false);
+  const [deleteTodoByID, setDeleteTodoByID] = useState<number | null>();
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(error => {
+        setErrorMessage(Errors.Load);
+        throw error;
+      });
+  }, []);
+
+  useEffect(() => {
+    if (errorMessage) {
+      setTimeout(() => setErrorMessage(null), 3000);
+    }
+  }, [errorMessage]);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [tempTodo, todos]);
+
+  const addNewTodo = () => {
+    const trimmedTitle = title.trim();
+
+    if (!trimmedTitle) {
+      setErrorMessage(Errors.EmptyTitle);
+
+      return;
+    }
+
+    setIsAddingTodo(true);
+
+    const temp = {
+      title: trimmedTitle,
+      id: 0,
+      completed: false,
+      userId: USER_ID,
+    };
+
+    setTempTodo(temp);
+    setErrorMessage(null);
+
+    postTodo(temp)
+      .then(res => {
+        setTodos(prev => [...prev, res]);
+        setTitle('');
+      })
+      .catch(() => setErrorMessage(Errors.Add))
+      .finally(() => {
+        setTempTodo(null);
+        setIsAddingTodo(false);
+      });
+  };
+
+  const deleteCurrentTodo = useCallback((id: number) => {
+    setDeleteTodoByID(id);
+    deleteTodo(id)
+      .then(() => {
+        setTodos(prev => prev.filter(todo => todo.id !== id));
+      })
+      .catch(() => {
+        setErrorMessage(Errors.Delete);
+      })
+      .finally(() => setDeleteTodoByID(null));
+  }, []);
+
+  const clearCompleted = async () => {
+    const todoToClear: Todo[] = [];
```

Comment:
Hmmm, let's try to simplify it a bit here: you can create an array with todoToClear by filtering current todos on completed status. 

Then, you can use Promise.all to run all deletion operations at once by mapping through todoToClear.

Also, it's worth having a toggling loading state logic here.